### PR TITLE
[DROOLS-6445] Remove old factories from org.kie.pmml.compiler.commons.factories

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola @jiripetrlik

See  https://issues.redhat.com/browse/DROOLS-6445

To sum up
1) removed usage of "old" kiepmmlpredicate from tree/node
2) removed usage of "old" kiepmmlpredicate from mining/segment
3) deleted unused factories and tests

### CODE REVIEW

**Comment by gitgabrio** [comment]
Jenkins run fdb

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/AbstractKiePMMLFactoryTest.java [hunk 1]
```diff
@@ -57,6 +60,8 @@ protected static void innerSetup() throws JAXBException, SAXException, IOExcepti
         MINING_MODEL = (MiningModel) pmml.getModels().get(0);
         assertNotNull(MINING_MODEL);
         populateMissingIds(MINING_MODEL);
+        DERIVED_FIELDS = getDerivedFields(TRANSFORMATION_DICTIONARY,
+                                                                  MINING_MODEL.getLocalTransformations());
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/AbstractKiePMMLFactoryTest.java]: Fix formatting

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
 
     public KiePMMLSegmentTemplate() {
-        super(name, Collections.emptyList(), kiePMMLPredicate, model);
+        super(name, Collections.emptyList(), getPredicate(), model);
         weight = 1;
         id = null;
     }
 
+    private static KiePMMLPredicate getPredicate() {
+        return null;
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl:24]: It seems that this method does nothing. It only returns null. Is it intentional?

**Review by jiripetrlik — APPROVED** [review]
LGTM. I have only one question.

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl:24]: Hi  @jiripetrlik 
This is just the template I use for the method signature. It could be empty or whatever - the body is populated during codegen

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactory.java [hunk 3]
```diff
@@ -91,7 +92,10 @@ public static KiePMMLMiningModel getKiePMMLMiningModel(final DataDictionary data
                                                                       final List<KiePMMLModel> nestedModels) {
         logger.trace("getKiePMMLMiningModelSourcesMap {} {} {}", dataDictionary, model, parentPackageName);
         final String segmentationName = String.format(SEGMENTATIONNAME_TEMPLATE, model.getModelName());
+        final List<DerivedField> derivedFields = getDerivedFields(transformationDictionary,
```
Comment (danielezonca) [inline kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactory.java:95]: Is this a fix?
It doesn't seem related to this PR 🤔 
If so, can you add an end to end test (aka based on real pmml file) to cover this change?

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl (hunk 2 above)
Comment (jiripetrlik) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-model/src/main/resources/KiePMMLSegmentTemplate.tmpl:24]: Understand. Thank you for the explanation.

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/AbstractKiePMMLFactoryTest.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/test/java/org/kie/pmml/models/mining/compiler/factories/AbstractKiePMMLFactoryTest.java]: @danielezonca done

File: kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactory.java (hunk 3 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-models/kie-pmml-models-mining/kie-pmml-models-mining-compiler/src/main/java/org/kie/pmml/models/mining/compiler/factories/KiePMMLMiningModelFactory.java:95]: @danielezonca 
derivedFields are needed to instantiate the predicate inside the segment.
I put that line here to be sure to have the same objects down all the code path.
I slightly modified MiningModel_Mixed.pmml to have a segment' predicate using a derived field

**Review by danielezonca — APPROVED** [review]
